src/main/java/stsjorbsmod/cards/wanderer/ThirstingSword.java

```diff
@@ -0,0 +1,82 @@
+package stsjorbsmod.cards.wanderer;
+
+import com.megacrit.cardcrawl.actions.AbstractGameAction;
+import com.megacrit.cardcrawl.actions.common.ApplyPowerAction;
+import com.megacrit.cardcrawl.actions.common.DamageAction;
+import com.megacrit.cardcrawl.actions.common.HealAction;
+import com.megacrit.cardcrawl.cards.DamageInfo;
+import com.megacrit.cardcrawl.characters.AbstractPlayer;
+import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
+import com.megacrit.cardcrawl.monsters.AbstractMonster;
+import com.megacrit.cardcrawl.powers.AbstractPower;
+import stsjorbsmod.JorbsMod;
+import stsjorbsmod.cards.CustomJorbsModCard;
+import stsjorbsmod.characters.Wanderer;
+import stsjorbsmod.powers.BurningPower;
+
+import static stsjorbsmod.JorbsMod.makeCardPath;
+
+/**
+ * BASE: Deal 10 damage. Apply 3 Burning. Heal for the amount of Burning on target. Lose 3 Max HP. Exhaust.
+ * UPGRADE: Deal 12 damage. Apply 6 Burning. Heal for the amount of Burning on target. Lose 2 Max HP. Exhaust.
+ */
+
+public class ThirstingSword extends CustomJorbsModCard {
+    public static final String ID = JorbsMod.makeID(ThirstingSword.class.getSimpleName());
+    public static final String IMG = makeCardPath("Manipulation_Rares/thirsting_sword.png");
+
+    private static final CardRarity RARITY = CardRarity.RARE;
+    private static final CardTarget TARGET = CardTarget.ENEMY;
+    private static final CardType TYPE = CardType.ATTACK;
+    public static final CardColor COLOR = Wanderer.Enums.WANDERER_CARD_COLOR;
+
+    private static final int COST = 1;
+    private static final int DAMAGE = 10;
+    private static final int BURNING = 3;
+    private static final int LOSE_MAX_HP = 3;
+    private static final int UPGRADE_DAMAGE = 2;
+    private static final int UPGRADE_BURNING = 3;
+    private static final int UPGRADE_LOSE_MAX_HP = -1;
+
+    public ThirstingSword() {
+        super(ID, IMG, COST, TYPE, COLOR, RARITY, TARGET);
+        damage = baseDamage = DAMAGE;
+        magicNumber = baseMagicNumber = BURNING;
+        metaMagicNumber = baseMetaMagicNumber = LOSE_MAX_HP;
+
+        exhaust = true;
+
+        tags.add(CardTags.HEALING);
+    }
+
+    @Override
+    public void use(AbstractPlayer p, AbstractMonster m) {
+        addToBot(new DamageAction(m, new DamageInfo(p, damage), AbstractGameAction.AttackEffect.FIRE)); //Deal Damage
+        addToBot(new ApplyPowerAction(m, p, new BurningPower(m, p, magicNumber))); //Apply Burning
+        addToBot(new AbstractGameAction() {
```

Comment:
I think this would have an issue where it would be removed by `GameActionManager.clearPostCombatActions` before the heal got queued if the `DamageAction` were to kill the last enemy. We discussed this a bit in the discord and agreed that reordering the card effects such that it went "burning -> heal ->-maxHp -> damage" would be a reasonable way to solve the ordering issue.

I think to implement that, you'd want to:

* move the `DamageAction` down to queue it after this custom action
* use `addToTop` instead of `addToBot` to queue the `HealAction`
* move the `decreaseMaxHealth` call into the custom action`

Reply:
Thanks for the feedback!  I updated the card accordingly.  My one concern is that it doesn't synergize as well with the Lust memory anymore as now the burn gained from the damage dealt at the end doesn't count towards the heal.  But that might be minor enough to not matter.  If so, then the card should be good to go.